File: frontend/desktop/lib/models/requests/set_new_password_request.dart
/// Body for POST /auth/set-new-password — the forced-password-change flow
/// (MustChangePassword was true, see ForceChangePasswordScreen). Unlike
/// ChangePasswordRequest this needs no currentPassword: the caller is
/// already authenticated with the SuperAdmin-assigned password.
class SetNewPasswordRequest {
  final String newPassword;
  final String confirmPassword;

  const SetNewPasswordRequest({required this.newPassword, required this.confirmPassword});

  Map<String, dynamic> toJson() => {
        'newPassword': newPassword,
        'confirmPassword': confirmPassword,
      };
}

File: backend/services/identity/eTicketing.Identity.Business/Auth/Dtos/SetNewPasswordRequest.cs
namespace eTicketing.Identity.Business.Auth;

/// <summary>Used by the forced-password-change flow (SuperAdmin set a staff/org account's
/// password directly — see AdminService.SetPasswordAsync — and MustChangePassword is true).
/// Unlike ChangePasswordRequest this needs no CurrentPassword: the caller is already
/// authenticated with the SuperAdmin-assigned password and is replacing it with their own.</summary>
public record SetNewPasswordRequest
{
    public string NewPassword { get; init; } = string.Empty;
    public string ConfirmPassword { get; init; } = string.Empty;
}
